sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,313 @@
+package com.azure.core.implementation.http.rest;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.StreamResponse;
+import com.azure.core.implementation.AccessibleByteArrayOutputStream;
+import com.azure.core.implementation.TypeUtil;
+import com.azure.core.implementation.serializer.HttpResponseDecoder;
+import com.azure.core.util.Base64Url;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.core.util.tracing.TracerProxy;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.nio.ByteBuffer;
+import java.util.EnumSet;
+import java.util.function.Consumer;
+
+public class SyncRestProxy extends RestProxyBase {
+    /**
+     * Create a RestProxy.
+     *
+     * @param httpPipeline    the HttpPipelinePolicy and HttpClient httpPipeline that will be used to send HTTP requests.
+     * @param serializer      the serializer that will be used to convert response bodies to POJOs.
+     * @param interfaceParser the parser that contains information about the interface describing REST API methods that
+     */
+    public SyncRestProxy(HttpPipeline httpPipeline, SerializerAdapter serializer, SwaggerInterfaceParser interfaceParser) {
+        super(httpPipeline, serializer, interfaceParser);
+    }
+
+    /**
+     * Send the provided request asynchronously, applying any request policies provided to the HttpClient instance.
+     *
+     * @param request the HTTP request to send
+     * @param contextData the context
+     * @return a {@link Mono} that emits HttpResponse asynchronously
+     */
+    public HttpResponse send(HttpRequest request, Context contextData) {
+        return httpPipeline.sendSync(request, contextData);
+    }
+
+    public HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[] args) throws IOException {
+        return createHttpRequestBase(methodParser, serializer, false, args);
+    }
```

Comment:
that's no different from async version.

Should `createHttpRequestBase` -> `protected createHttpRequest` in the base class ?